File: backend/schemas/engagement.py
from pydantic import BaseModel, Field
from datetime import datetime


class CommentCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    message: str = Field(..., min_length=1, max_length=1000)


class CommentResponse(BaseModel):
    id: int
    name: str
    message: str
    created_at: datetime

    class Config:
        from_attributes = True


class PaginatedComments(BaseModel):
    items: list[CommentResponse]
    total: int
    page: int
    page_size: int


class ViewCountResponse(BaseModel):
    total_views: int
    today_views: int
    unique_visitors: int


class ViewTrackRequest(BaseModel):
    page: str = "/"

File: frontend/src/types/engagement.ts
export interface Comment {
  id: number;
  name: string;
  message: string;
  created_at: string;
}

export interface PaginatedComments {
  items: Comment[];
  total: number;
  page: number;
  page_size: number;
}

export interface ViewStats {
  total_views: number;
  today_views: number;
  unique_visitors: number;
}
